### PR TITLE
6 create base map unit tests

### PR DESCRIPTION
**Pull request by Todor-Shishmanov**
The tests and header for the basic implementation of Room. 
Sadly implementation also exists, ignored it for now. The next feature will be with developed with the new test-driven workflow, I swear!

For now, there will be no Map. (lazy developer)

Notice it is merging in 6-create-base-map.

### CODE REVIEW

**Review by mirela-manoleva — CHANGES_REQUESTED** [review]

File: Yoda/test_map.cpp [hunk 1]
```diff
@@ -12,43 +12,68 @@ namespace Yoda
...
+			}
+			catch (const std::invalid_argument& e) {
+				if (e.what() == std::string("Invalid layout: Layout must contain only unique IDs.")) return; // this passes the test.
+			}
+			catch (...) { Assert::Fail(L"Error in Room::Room() - Wrong exception thrown."); }
+			Assert::Fail(L"Error in Room::Room() - No exception thrown.");
+		}
 		TEST_METHOD(absolute_place_valid)
 		{
 			Logger::WriteMessage(L"Testing absolute_place() with valid input.");
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 2]
```diff
@@ -12,43 +12,68 @@ namespace Yoda
...
 			Assert::IsTrue(placed, L"Error in function Room::absolute_place(): Did not return true.");
-			Assert::AreEqual(newID, new_layout.at(newIDCell), L"Error in function Room::absolute_place(): ID was not placed in the cell.");
+			Assert::AreEqual(validID, layout.at(validCell), L"Error in function Room::absolute_place(): ID was not placed in the cell.");
 		}
 
+
 		TEST_METHOD(absolute_place_invalid_cell)
 		{
-			Logger::WriteMessage(L"Testing absolute_place() with invalid input - Invalid coordinates:");
+			Logger::WriteMessage(L"Testing absolute_place() with invalid input - Invalid Cell.");
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 3]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+			Assert::IsTrue(layout.at(removedCell) == NULL_OBJECT_ID, L"Error in function Room::remove(): The ID wasn't removed.");
+		}
+
+		TEST_METHOD(remove_invalid_id_1)
+		{
+			Logger::WriteMessage(L"Testing Room::remove() with invalid input - Removing NULL_OBJECT_ID");
+			Room room(layout_);
+			ID invalidID = NULL_OBJECT_ID;
+
+			bool isRemoved = room.remove(invalidID);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 4]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		}
+
+		TEST_METHOD(remove_valid)
+		{
+			Logger::WriteMessage(L"Testing Room::remove() with valid input.");
+			Room room(layout_);
+			ID validID = 1;
+			Cell removedCell(1, 2); //check layout_
+
+			bool isRemoved = room.remove(validID);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 5]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		TEST_METHOD(move_valid)
+		{
+			Logger::WriteMessage(L"Testing Room::move() with valid input.");
+			Room room(layout_);
+			ID validID = 1;
+			Direction validDirection = Direction::Forward; //check layout_
+			Cell originalCell(1, 2);
+			Cell forwardCell(1, 1);
+
+			bool isMoved = room.move(validID, validDirection);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 6]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		}
+
+		TEST_METHOD(move_invalid_direction_1)
+		{
+			Logger::WriteMessage(L"Testing Room::move() with invalid input - Outside of layout.");
+			Room room(layout_);
+			ID validID = 1;
+			Direction invalidDirection = Direction::Backward; //check layout_
+
+			bool isMoved = room.move(validID, invalidDirection);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 7]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		}
+
+		TEST_METHOD(move_invalid_direction_2)
+		{
+			Logger::WriteMessage(L"Testing Room::move() with invalid input - Cell is occupied by another ID.");
+			Room room(layout_);
+			ID validID = 1;
+			Direction invalidDirection = Direction::Right; //check layout_
+
+			bool isMoved = room.move(validID, invalidDirection);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 8]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		}
+
+		TEST_METHOD(move_invalid_id_1)
+		{
+			Logger::WriteMessage(L"Testing Room::move() with invalid input - ID is NULL_OBJECT_ID.");
+			Room room(layout_);
+			ID invalidID = NULL_OBJECT_ID;
+			Direction validDirection = Direction::Forward; //check layout_
+
+			bool isMoved = room.move(invalidID, validDirection);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

File: Yoda/test_map.cpp [hunk 9]
```diff
@@ -73,17 +97,309 @@ namespace Yoda
...
+		}
+
+		TEST_METHOD(move_invalid_id_2)
+		{
+			Logger::WriteMessage(L"Testing Room::move() with invalid input - Moving ID that doesn't exist");
+			Room room(layout_);
+			ID invalidID = 9;
+			Direction validDirection = Direction::Forward; //check layout_
+
+			bool isMoved = room.move(invalidID, validDirection);
```
[the remark this answers is not in the document]
Comment (Todor-Shishmanov) [reply Yoda/test_map.cpp]: Resolved. Accepted suggestion.

**Pull request merged**